File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        l,r=0,0
        highest=0

        for r in range(len(prices)):
            while prices[r] < prices[l]:
                l += 1

            if prices[r] > prices[l]:
                highest=max(highest,prices[r]-prices[l])
        return highest







        





            

    


        
            
        


        

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cs
public class Solution {
    public int MaxProfit(int[] prices) {

        int highest=0;
        int l=0;
        for(int i=0;i<prices.Length;i++){
            while(prices[l] > prices[i]){
                l ++;
            }
            if(prices[i] > prices[l]){
                highest=Math.Max(highest,prices[i]-prices[l]);
            }
        }
        return highest;


        
    }
}
